components/nimbus/android/src/main/java/org/mozilla/experiments/nimbus/internal/FeatureHolder.kt

```diff
@@ -1,33 +1,47 @@
 package org.mozilla.experiments.nimbus.internal
 
-import android.content.Context
 import org.mozilla.experiments.nimbus.FeaturesInterface
 import org.mozilla.experiments.nimbus.Variables
 import org.mozilla.experiments.nimbus.NullVariables
+import java.util.concurrent.locks.ReentrantLock
 
+/**
+ * `FeatureHolder` is a class that unpacks a JSON object from the Nimbus SDK and transforms it into a useful
+ * type safe object, generated from a feature manifest (a `.fml.yaml` file).
+ *
+ * The two routinely useful methods are the `value()` and `recordExposure()` events.
+ *
+ * There are methods useful for testing, and more advanced uses: these all start with `with`.
+ */
 class FeatureHolder<T>(
     private val getSdk: () -> FeaturesInterface?,
     private val featureId: String,
-    private val create: (Variables) -> T
+    private var create: (Variables) -> T
 ) {
+    private val lock = ReentrantLock()
+
+    private var cachedValue: T? = null
 
     /**
      * Get the JSON configuration from the Nimbus SDK and transform it into a configuration object as specified
      * in the feature manifest. This is done each call of the method, so the method should be called once, and the
      * result used for the configuration of the feature.
      *
-     * An optional `Context` object is taken which is used to look up resources. Most of the time this isn't required, and the context can be
-     * derived from the `Nimbus` singleton object. This is now deprecated, and will be removed in future releases.
-     *
      * @returns T
-     * @throws NimbusFeatureException thrown before the Nimbus object has been constructed or `FxNimbus.api` has not been set.
-     * This can be resolved by setting `FxNimbus.api`, and after that by passing in a `Context` object.
+     * @throws NimbusFeatureException thrown before the Nimbus object has been constructed or `FxNimbus.initialize` has not been set.
+     * This can be resolved by setting `FxNimbus.initialize`, and after that by passing in a `Context` object.
      */
-    @Suppress("UNUSED_PARAMETER")
-    fun value(context: Context? = null): T {
-        val variables = getSdk()?.getVariables(featureId, false) ?: NullVariables.instance
-        return create(variables)
-    }
+    fun value(): T =
```

Comment:
This is a breaking change correct? do you know if Fenix already calls this without the argument? Otherwise we might want to mark this as a breaking change (or keep the parameter)

Reply:
Now that the AS 93.0.1 has been backed out, I should put the unused parameter back.

I'd prefer to do this with the three PR doctrine we talked about.